### PR TITLE
Added compatibility for fully qualified names in Javadoc in TypeReferenceRefactor.

### PR DESCRIPTION
**Pull request by 10wonga**
testChangeTypes() now passes and includes fully qualified name types as the issue mentioned.

### CODE REVIEW

File: astra-core/src/main/java/org/alfasoftware/astra/core/refactoring/operations/types/TypeReferenceRefactor.java [hunk 1]
```diff
@@ -201,7 +201,16 @@ public boolean visit(Javadoc node) {
               .map(ITypeBinding::getQualifiedName)
               .filter(n -> n.equals(getFromType()))
               .ifPresent(t -> types.add((SimpleName) f));
-          } else if (f instanceof MethodRef) {
+          }
+          else if (f instanceof QualifiedName) {
+            Optional.of(f)
+              .map(QualifiedName.class::cast)
+              .map(QualifiedName::resolveTypeBinding)
```
Comment (RadikalJin) [inline astra-core/src/main/java/org/alfasoftware/astra/core/refactoring/operations/types/TypeReferenceRefactor.java:208]: Do we need to resolve this type binding?

**Comment by RadikalJin** [comment]
Great stuff, thanks @10wonga. Just looking into the changes now - think there’s an unnecessary binding resolve but otherwise all looks good.

File: astra-core/src/main/java/org/alfasoftware/astra/core/refactoring/operations/types/TypeReferenceRefactor.java (hunk 1 above)
Comment (10wonga) [reply astra-core/src/main/java/org/alfasoftware/astra/core/refactoring/operations/types/TypeReferenceRefactor.java:208]: nope, I don't think there is a need for this as there would be no information for a method's binding type.
will remove line 208 now

**Comment by RadikalJin** [comment]
I’m not sure this is correct - is the qualifier the package name? It’d be great to check for this with another test that updates references to a new package, but with the same simple name. Eg. update a javadoc reference from
com.Foo
to
com.bar.Foo

The test could also include Javadoc references to 
com.Bar
and
com.other.Foo
Which should not be changed.

**Comment by 10wonga** [comment]
Ah yes I do think I'm on the same page now, after checking the issue description again I have identified that the change needed is to use _getFullyQualifiedName_

**Review by 10wonga — COMMENTED** [review]
Latest attempt at resolving else if loop line 205

**Comment by RadikalJin** [comment]
Excellent, thanks @10wonga. I've added some negative tests to check that we don't refactor where we don't find a match - and these tests have passed. Merging in your changes now.

**Pull request merged**